common/src/main/java/org/vivecraft/server/config/ConfigBuilder.java

```diff
@@ -225,32 +228,111 @@ public T reset() {
             return defaultValue;
         }
 
+        public boolean isDefault() {
+            return Objects.equals(get(),  defaultValue);
+        }
+        public String getComment() {
+            return config.getComment(path);
+        }
+
         public String getPath() {
             return String.join(".", path);
         }
     }
 
     public static class BooleanValue extends ConfigValue<Boolean>{
-        public BooleanValue(Config config, List<String> path, boolean defaultValue) {
+        public BooleanValue(CommentedConfig config, List<String> path, boolean defaultValue) {
             super(config, path, defaultValue);
         }
     }
 
     public static class StringValue extends ConfigValue<String>{
-        public StringValue(Config config, List<String> path, String defaultValue) {
+        public StringValue(CommentedConfig config, List<String> path, String defaultValue) {
+            super(config, path, defaultValue);
+        }
+    }
+
+    public static class ListValue<T> extends ConfigValue<List<T>>{
+        public ListValue(CommentedConfig config, List<String> path, List<T> defaultValue) {
             super(config, path, defaultValue);
         }
     }
 
-    public static class IntValue extends ConfigValue<Integer>{
-        public IntValue(Config config, List<String> path, int defaultValue) {
+    public static class InListValue<T> extends ConfigValue<T>{
+        private final Collection<? extends T> validValues;
+        public InListValue(CommentedConfig config, List<String> path, T defaultValue, Collection<? extends T> validValues) {
+            super(config, path, defaultValue);
+            this.validValues = validValues;
+        }
+
+        public Collection<? extends T> getValidValues(){
+            return validValues;
+        }
+    }
+
+    public interface NumberValue<E extends Number>{
```

Comment:
Can instead of an interface you not use a base class instead?

Reply:
fixed